Source/WebCore/Modules/reporting/ReportingObserver.cpp

```diff
@@ -53,15 +77,52 @@ ReportingObserver::~ReportingObserver()
 
 void ReportingObserver::disconnect()
 {
+    m_reporting->unregisterReportingObserver(*this);
 }
 
 void ReportingObserver::observe()
 {
+    m_reporting->registerReportingObserver(*this);
```

Comment:
Who keeps the observer alive?
Otherwise, we'd just GC this observer at any moment.

Reply:
The reporting observer is created in JS by the page author. I assumed the lifecycle would be controlled by the script or page lifetime, but perhaps this is not right? Maybe it should be an ActiveDOMObject?

Reply:
That's not quite right. So long as observer is observing anything at all, it's possible for the observer to be called back in the future. In effect, it needs to be kept alive so long as the global object is alive. As we discussed below, using `GenerateIsReachable=ImplScriptExecutionContext` will do the trick.